test/test_s3_utils.py

```diff
@@ -11,13 +12,55 @@
 from unittest import mock
 
 
+@contextlib.contextmanager
+def mocked_s3_integration(integrated_names=None, zip_suffix="", ffenv=None):
+    """
+    This does common setup of some mocks needed by zip testing.
+    """
+
+    zip_path_key = "zip_path" + zip_suffix
+    zip_filename_key = "zip_filename" + zip_suffix
+
+    b3 = MockBoto3()
+
+    if not ffenv:
+        ffenv = integrated_names['ffenv'] if integrated_names else None
+
+    with mock.patch.object(s3_utils_module, "boto3", b3):
+
+        s3_connection = s3Utils(env=ffenv)
+
+        if integrated_names is not None:
+
+            # Not needed when mocked.
+            # s3_connection.s3_delete_dir(prefix)
+
+            # In our mock, this won't exist already on S3 like in the integrated version of this test,
+            # so we have to pre-load to our mock S3 manually. -kmp 13-Jan-2021
+            s3_connection.s3.upload_file(Filename=integrated_names[zip_path_key],
+                                         Bucket=s3_connection.outfile_bucket,
+                                         Key=integrated_names[zip_filename_key])
+
+            s3_connection.s3.put_object(Bucket=s3_connection.outfile_bucket,
+                                        Key=integrated_names['filename'],
+                                        Body=str.encode('thisisatest'))
+
+        yield s3_connection
+
+
+@pytest.mark.integrated
 @pytest.mark.parametrize('ff_ordinary_envname', ['fourfront-mastertest', 'fourfront-webdev', 'fourfront-hotseat'])
 def test_s3utils_creation_ff_ordinary(ff_ordinary_envname):
     util = s3Utils(env=ff_ordinary_envname)
     assert util.sys_bucket == 'elasticbeanstalk-%s-system' % ff_ordinary_envname
 
 
+@pytest.mark.integrated
 def test_s3utils_creation_ff_stg():
+    # TODO: I'm not sure what this is testing, so it's hard to rewrite
+    #   But I fear this use of env 'staging' implies the GA test environment has overbroad privilege.
+    #   We should make this work without access to 'staging'.
+    #   -kmp 13-Jan-2021
```

Comment:
It's just testing that s3utils resolves the s3 buckets correctly. It also tests resolving staging from `get_standard_mirror_env(compute_ff_prd_env())`.

Reply:
ok. it's probably possible to mock that. i was worried it was trying to test that particular files are in place on specific machines or that those files are in a particular form. that's something that can change dynamically and not in this code, and that mocking would not be able to test. I won't fix that for this PR but can work with it in the future.